contracts/oracle/L2ScalingPriceOracle.sol

```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import {Decimal} from "../external/Decimal.sol";
+import {Constants} from "./../Constants.sol";
+import {Deviation} from "./../utils/Deviation.sol";
+import {ScalingPriceOracle} from "./ScalingPriceOracle.sol";
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {Initializable} from "@openzeppelin/contracts/proxy/utils/Initializable.sol";
+
+/// @notice contract that receives a chainlink price feed and then linearly interpolates that rate over
+/// a 28 day period into the VOLT price. Interest is compounded monthly when the rate is updated
+/// Specifically built for L2 to allow a deployment that is mid-month
+/// @author Elliot Friedman
+contract L2ScalingPriceOracle is ScalingPriceOracle {
```

Comment:
What's the need for this contract, and how is it any different than the mainnet scaling price oracle?

Reply:
The reason this contract is that when we deploy on Arbitrum, we need to be able to sync with the mainnet oracle even if we don't deploy it at the exact same time that we request the monthly data from chainlink.

So let's say we deploy on the 25th of the month, now it messes up when we can request data as the timer counts down 28 days since creation until we can request chainlink data. So if we don't set the start time for the Timed class in the past, we have to wait until the 26th of the next month to request CPI data that came out on the 10th of the month, and that the mainnet scaling price oracle received on the 15th. This means the system target price gets desynchronized across chains potentially opening up arbitrage loops by which PCV could be drained if the spread widened enough.

Reply:
Essentially this contract lets us set the oracle starting price `oraclePrice` that was the last cached price on mainnet and the timed class `startTime` which was the last time oracle data was requested on mainnet.

With these two variables being set the exact same as mainnet, then our system is perfectly synced up and has the same price across mainnet and L2.